install/kubernetes/helm/istio/templates/_affinity.tpl

```diff
@@ -40,4 +40,51 @@
           - {{ $key }}
     {{- end }}
   {{- end }}
-{{- end }}
\ No newline at end of file
+{{- end }}
+
+{{- define "podAntiAffinity" }}
+{{- if or .Values.podAntiAffinityLabelSelector .Values.podAffinityTermLabelSelector}}
+  podAntiAffinity:
+    {{- if .Values.podAntiAffinityLabelSelector }}
+    requiredDuringSchedulingIgnoredDuringExecution:
+    {{- include "podAntiAffinityRequiredDuringScheduling" . }}
+    {{- end }}
+    {{- if or .Values.podAffinityTermLabelSelector}}
+    preferredDuringSchedulingIgnoredDuringExecution:
+    {{- include "podAntiAffinityPreferredDuringScheduling" . }}
+    {{- end }}
+{{- end }}
+{{- end }}
+
+{{- define "podAntiAffinityRequiredDuringScheduling" }}
+      {{- range $key, $val := .Values.podAntiAffinityLabelSelector }}
+      - labelSelector:
+        matchExpressions:
+        - key: {{ $key }}
+          operator: In
+          values:
+          - {{ $val }}
+        topologyKey: "kubernetes.io/hostname"
```

Comment:
Not sure if we can parameterize these `topologykey`, but it's a little complicated as I read the rules here: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#an-example-of-a-pod-that-uses-pod-affinity

Reply:
yes, it is complicated, think this is a starting point as recommended from #11333